### ISSUE TITLE
Notification widgets missing border in high contrast

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#7562]
## Environment

> Operating system

Windows 10

> Browser

Edge

## Detailed description

> What version of the Carbon Design System are you using?

10.26

> What did you expect to happen?

I thought there would be a border around the inline and toast notifications, like this:

<img width="1028" alt="Screen Shot 2021-01-14 at 10 45 42" src="https://user-images.githubusercontent.com/69599/104533402-abd4f000-5655-11eb-8097-0db54bebf2fc.png">

> What happened instead?

No border:

<img width="785" alt="Screen Shot 2021-01-14 at 10 41 06" src="https://user-images.githubusercontent.com/69599/104533331-87791380-5655-11eb-8315-7d6bb698ff19.png">

<img width="576" alt="Screen Shot 2021-01-14 at 10 42 53" src="https://user-images.githubusercontent.com/69599/104533338-8b0c9a80-5655-11eb-9ca1-f7b11abbe1d9.png">

## Steps to reproduce the issue

Just try https://react.carbondesignsystem.com/?path=/story/notifications--inline and https://react.carbondesignsystem.com/?path=/story/notifications--toast on Edge or Firefox, on a Windows machine set to high contrast.

## Additional information

We worked around this with SCSS of

```
.d-hc {
	.#{$prefix}--inline-notification {
		border-style: solid;
		border-width: 1px 1px 1px 3px;
		@at-root .d-rtl#{&} {
			border-width: 1px 3px 1px 1px;
		}
		&:before {
			border: none;
		}
	}
}
```

Yours will be different since you use a `@media` query for detecting high contrast, and since you don't manually add RTL CSS.

**Comment by tw15egan** [comment]
Should be closed via https://github.com/carbon-design-system/carbon/pull/7601
